br/pkg/lightning/importer/chunk_process.go

```diff
@@ -362,6 +366,7 @@ func (cr *chunkProcessor) encodeLoop(
 						if err = t.initializeColumns(columnNames, cr.chunk); err != nil {
```

Comment:
`cr.chunk.ColumnPermutation` is still filled and stored into checkpoint here

Reply:
maybe we can fix it here https://github.com/pingcap/tidb/blob/aab2da047904901725e5a3bf745b47b1488eb8c7/br/pkg/lightning/importer/chunk_process.go#L102-L104

when there's extend column, no need to set columns

Reply:
It's OK to store them into checkpoint, but it must not to add extended columns again after loading from checkpoint

Reply:
the ColumnPermutation is correct, the problem here is that we have 1 more column generated after restart, see code link above